Sources/Swinject/ServiceEntry.swift

```diff
@@ -99,3 +99,13 @@ public final class ServiceEntry<Service>: ServiceEntryProtocol {
         )
     }
 }
+
+// Concreate type for factories stored inside of Swinject ServiceEntry.
+// This avoids some limitations in the Swift generic type system by wrapping the closure in a struct
+internal struct ServiceFactory<each Argument> {
```

Comment:
I'm getting `Parameter packs in generic types are only available in iOS 17.0.0 or newer` when building cash-ios